machine/machine_core/ssh_connection.py

```diff
@@ -69,6 +69,22 @@ def message(self, *args):
         if self.verbose:
             sys.stderr.write(" ".join(args) + '\n')
 
+    def print_console_log(self):
+        """Prints VM's console to stderr"""
+        console_file = "/tmp/console.log"
+        try:
+            f = open(console_file)
+            console_log = f.read().strip()
+
+            if len(console_log) == 0:
+                sys.stderr.write(f"VM's console log file '{console_file}' is empty")
+                return
+
+            sys.stderr.write(f"---- Console log start's here ----\n{console_log}\n---- Console log end's here ----")
```

Comment:
"starts" and "ends" (no apostrophes)

Also, write() does not automatically append a `\n`. Either do that, or use `print(..., file=sys.stderr)`

Reply:
Fixed